api/Application.Core/Data/DataExtensions.cs

```diff
@@ -62,7 +62,43 @@ public static IPageableList<TSource> ToPageableList<TSource>(this IOrderedQuerya
 			};
 		}
 
-		public static void Add<TModel, TDataContext>(this TDataContext dataContext, TModel model)
+        public static async Task<IPageableList<TSource>> ToPageableListAsync<TSource>(this IOrderedQueryable<TSource> source, int pageNumber = 1, int pageSize = 0)
+        {
+            var totalCount = await source.CountAsync();
+
+            var pageIndex = Math.Max(0, pageNumber - 1);
+
+			// Hangs if ToListAsync is used
+            var pageItems = await Task.Run(() =>
+            {
+                return source
```

Comment:
Can we use ToListAsync()? Please avoid using Task.Run

Reply:
If we use ToListAsync() method freezes after this line is executed. Shall i use ToList() without Task.Run then?